test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -12480,6 +12481,38 @@ def forward(self, x):
 
         self.run_test(LerpModel(), torch.rand(5, 4, 3))
 
+    @common_utils.parametrize("use_complex_dtypes", [True, False])
+    @skipIfUnsupportedMinOpsetVersion(9)
+    def test_print_tensor_within_torch_nn_module(self, use_complex_dtypes: bool):
+        class PrintTensorOnMyModel(torch.nn.Module):
+            def forward(self, x):
+                x_firsts = x[:, 0]
+                print(f"x_firsts: {x_firsts}")
+                return x_firsts
+
+        m = PrintTensorOnMyModel().eval()
+        if use_complex_dtypes:
+            dtype = torch.cfloat
+        else:
+            dtype = torch.float
+        x = torch.randn(10, 5, dtype=dtype)
+
+        if use_complex_dtypes:
+            if GLOBALS.onnx_shape_inference:
+                exception_type = RuntimeError
+            else:
+                exception_type = errors.OnnxExporterError
```

Comment:
OnnxExporterError is a RuntimeError so checking for RuntimeError is good enough I think

Reply:
This is an implementation detail that does not need to leak

Reply:
`errors.OnnxExporterError` intentionally subclassed from RuntimeError so users don't need to care about the custom error types we define when they don't want to. They provide more details to users for context. I just think catching two different errors that are not at the same level (OnnxExporterError is one level down) can be confusing. Ideally, we should have the checker raise OnnxExporterError too.

As an alternative for readability, I would break the test into two because having control flow in tests are generally to be avoided (for them to be more robust).

In any case, I suggest we merge our work to #86180